src/Libraries/Microsoft.Extensions.AI.Abstractions/Tools/HostedMcpServerTool.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Shared.Diagnostics;
+
+namespace Microsoft.Extensions.AI;
+
+/// <summary>
+/// Represents a hosted MCP server tool that can be specified to an AI service.
+/// </summary>
+public class HostedMcpServerTool : AITool
+{
+    /// <summary>
+    /// Initializes a new instance of the <see cref="HostedMcpServerTool"/> class.
+    /// </summary>
+    /// <param name="serverName">The name of the remote MCP server.</param>
+    /// <param name="url">The URL of the remote MCP server.</param>
+    public HostedMcpServerTool(string serverName, Uri url)
```

Comment:
We might want a convenience constructor that accepts a string url and just delegates to this